src/client/app/containers/BarChartContainer.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import _ from 'lodash';
+import { Bar } from 'react-chartjs-2';
+import moment from 'moment';
+import { connect } from 'react-redux';
+import 'chartjs-plugin-zoom';
+
+/**
+ * @param {State} state
+ */
+function mapStateToProps(state) {
+	const timeInterval = state.graph.timeInterval;
+	const barDuration = state.graph.barDuration;
+	const data = { datasets: [] };
+	// getColor() cycles through the colors, wrapping around the end to the beginning
+	const colors = ['LightBlue', 'GoldenRod', 'Black', 'OrangeRed', 'LightSeaGreen', 'LightSlateGray', 'Purple'];
+	let colorPointer = 0;
+	function getColor() {
+		const color = colors[colorPointer];
+		colorPointer = (colorPointer + 1) % colors.length;
+		return color;
+	}
+
+	const labelsSet = new Set();
+	for (const meterID of state.graph.selectedMeters) {
+		const readingsData = state.readings.bar.byMeterID[meterID][timeInterval][barDuration];
+		if (readingsData !== undefined && !readingsData.isFetching) {
+			const color = getColor();
+			data.datasets.push({
+				label: state.meters.byMeterID[meterID].name,
+				data: readingsData.readings.map(arr => arr[1]),
+				backgroundColor: color,
+				hoverBackgroundColor: color
+			});
+			// Add only the unique time intervals to the label set
+			for (const element of _.flatten(readingsData.readings.map(arr => arr[0]))) {
+				labelsSet.add(`${moment(element).format('MMM DD, YYYY, hh:mm a')} - ${moment(element).add(moment.duration(barDuration)).format('MMM DD, YYYY, hh:mm a')}`);
```

Comment:
I disagree with you here. It makes perfect sense for this line, but a couple more lines down, where I do `data.labels = Array.from(labelsSet).sort((x, y) => moment(x.split(' - ')[0], 'MMM DD, YYYY, hh:mm a').format('x') - moment(y.split(' - ')[0], 'MMM DD, YYYY, hh:mm a').format('x'));`
, it really complicates things.